packages/flutter_tools/lib/src/commands/build_ios_framework.dart

```diff
@@ -23,26 +25,21 @@ import '../runner/flutter_command.dart' show DevelopmentArtifact, FlutterCommand
 import '../version.dart';
 import 'build.dart';
 
-/// Produces a .framework for integration into a host iOS app. The .framework
-/// contains the Flutter engine and framework code as well as plugins. It can
-/// be integrated into plain Xcode projects without using or other package
-/// managers.
-class BuildIOSFrameworkCommand extends BuildSubCommand {
-  BuildIOSFrameworkCommand({
+abstract class BuildFrameworkCommand extends BuildSubCommand {
+  BuildFrameworkCommand({
     // Instantiating FlutterVersion kicks off networking, so delay until it's needed, but allow test injection.
     @visibleForTesting FlutterVersion? flutterVersion,
     required BuildSystem buildSystem,
     required bool verboseHelp,
     Cache? cache,
-    Platform? platform
+    Platform? platform,
   }) : _injectedFlutterVersion = flutterVersion,
        _buildSystem = buildSystem,
        _injectedCache = cache,
        _injectedPlatform = platform,
        super(verboseHelp: verboseHelp) {
     addTreeShakeIconsFlag();
     usesTargetOption();
-    usesFlavorOption();
```

Comment:
macOS doesn't support flavors, only add option in subclass `BuildIOSFrameworkCommand`